src/components/ProblemsPage/Selection.tsx

```diff
@@ -1,26 +1,62 @@
-import * as React from 'react';
+import React, { useEffect, useState } from 'react';
+import {
+  UseRefinementListProps,
+  useInstantSearch,
+  useRefinementList,
+} from 'react-instantsearch';
 import Select from 'react-select';
 import { useDarkMode } from '../../context/DarkModeContext';
 
-export default function Difficulty({ items, refine }) {
-  const darkMode = useDarkMode();
-
-  const handleChange = e => {
-    const refinements = [];
-    for (const difficulty of e) {
-      refinements.push(difficulty.label);
-    }
-    refine(refinements);
-  };
+export type SelectionProps = UseRefinementListProps & {
+  placeholder: string;
+  searchable: boolean;
+  isMulti: boolean;
+  transformLabel?: (label: string) => string;
+  items?: { label: string; value: string | string[] }[];
+};
 
+export default function Selection({
+  attribute,
+  limit,
+  placeholder,
+  searchable,
+  isMulti,
+  transformLabel: transform,
+  items,
+  ...props
+}: SelectionProps) {
+  const { items: refineItems } = useRefinementList({
+    attribute,
+    limit,
+    ...props,
+  });
+  if (!items) items = refineItems;
+  const darkMode = useDarkMode();
+  const [refinements, setRefinements] = useState<string[]>([]);
+  const { setIndexUiState } = useInstantSearch();
+  useEffect(() => {
+    setIndexUiState(prevIndexUiState => ({
+      refinementList: {
+        ...prevIndexUiState.refinementList,
+        [attribute]: refinements,
+      },
+    }));
+  }, [refinements]);
```

Comment:
on page load, will this be called once per `<Selection>` component? If so, will this result in many "operations"?

Reply:
Hm, I'm not sure since refinements starts as an empty array; not sure if that counts towards operations? Is there any API to track operations (or for example log whenever an "operation" occurs)?

Reply:
Maybe try monitoring the chrome network tab? Or try googling, I vaguely recall reading some post about how to do this...

Reply:
I think according to [this algolia page](https://support.algolia.com/hc/en-us/articles/4406975243921-What-is-an-operation-)
> If you use the [disjunctive faceting](https://www.algolia.com/doc/guides/managing-results/refine-results/faceting/#conjunctive-and-disjunctive-faceting) feature in your search, the number of operations for each query depends on the number of disjunctive facets you use.

It seems like each filter we use counts as an operation? (so I guess even with `setIndexUiState`, we're still taking 1000 operations 😭)

Reply:
if the only issue is filtering by status, can we disable it for now and merge the rest if it's working?

Reply:
Hm in that case the remaining refinement lists are still a bit unoptimized; apparently if you call `refine` on an already refined filter, it unrefines that filter (which isn't really mentioned anywhere in the docs). So in our current case right now if a user already has 2 refinements active and adds a 3rd one, this will take us 3 operations (i believe?), but theoretically it should only take 1. However, I'm still not quite sure how `setIndexUiState` counts operations yet, so I'm just waiting for algolia comes back online to verify 😓